File: W2/Valid Parentheses.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<char>a;
        for(char i:s){
            if(i=='(' || i=='{' || i=='[')
            a.push(i);
            else{
                if(a.empty()) return false;
                if(i==')' && a.top()=='(' || i==']' && a.top()=='[' || i=='}' && a.top()=='{')
                a.pop();
                else
                return false;
            }
        }
        return a.empty();
    }
};

File: W2/Valid Parentheses.dart
class Solution {
  bool isValid(String s) {
    List<String> a=[] ;
    if(s.length==0) return false;

    for(int i=0;i<s.length;i++){
      if(s[i]=='{' || s[i]=='(' || s[i]=='[' ){
        a.add(s[i]);
      }
      else{
        if(a.length !=0 &&(s[i]=='}'&&a.last=='{' || s[i]==')'&&a.last=='(' || s[i]==']'&&a.last=='[')){
          a.removeLast();
        }
        else{
          return false;
        }
      }
    }
    return a.isEmpty;
  }
}

void main(List<String> args) {
  Solution s1 =Solution();
  String s = "()[]{}";
  print(s1.isValid(s));
}